addons/sale_product_configurator/controllers/main.py

```diff
@@ -253,81 +169,193 @@ def _get_product_information(
+            'price': pricelist._get_product_price(
                 product_or_template.with_context(
                     **product_or_template._get_product_price_context(combination)
                 ),
                 **kwargs,
             ),
-        )
+        })
+        return super_dict
+
+    def _get_product_information_additional_params(self, currency_id=None, pricelist_id=None, so_date=None, **kwargs):
+        """ Return a dict of additional parameters to be given to `_get_product_information`.
+
+        For a module to customize the parameters given to `_get_product_information`, it must override this
+        method and call super, then add the parameters.
+
+        :param dict kwargs: Additional parameters used to compute the additionla params
+        :rtype: dict
+        :return: additional parameters
+        """
+        return {
+            **super()._get_product_information_additional_params(
+                currency_id=currency_id,
+                pricelist_id=pricelist_id,
+                so_date=so_date,
+                **kwargs
+            ),
+            'currency_id': currency_id,
+            'pricelist_id': pricelist_id,
+            'so_date': so_date,
+        }
+
+    def _get_optional_product(self, product_template, combination, only_main_product, **kwargs):
+        """ Return a dict of optional products based on a product_template and a combination.
+
+        Here, as the field 'optional_product_ids' does not exist on the product_template
+        model, this method will return an empty array ans will be overrided where needed.
+
+        :param recordset product_template: The product for which to seek information, as a
+                                        `product.template`.
+        :param recordset combination: The combination of the product, as a
+                                      `product.template.attribute.value` recordset.
+        :param bool only_main_product: Whether the optional products should be included or not.
+        :param dict kwargs: Additional parameters used to compute the additionla params
+        :rtype: dict
+        :return: optional products
+        """
+        return [
+            dict(
+                **self._get_product_information(
+                    optional_product_template,
+                    optional_product_template._get_first_possible_combination(
+                        parent_combination=combination
+                    ),
+                    # giving all the ptav of the parent product to get all the exclusions
+                    parent_combination=product_template.attribute_line_ids.\
+                        product_template_value_ids,
+                    **self._get_product_information_additional_params(**kwargs),
+                ),
+                parent_product_tmpl_ids=[product_template.id],
+            ) for optional_product_template in product_template.optional_product_ids
+        ] if not only_main_product else []
```

Comment:
```suggestion
        ] 
```

Reply:
Why would we get the optional products if ``only_main_product`` is true? Optional products do not equal to optional ones?